bin/rule-generator.js

```diff
@@ -0,0 +1,448 @@
+			},
+			{
+				name: `${ruleName}.json`,
+				content: JSON.stringify(
+					{
+						description: `${ruleName} tests`,
+						rule: `${ruleName}`,
+						violations: [],
+						passes: []
+					},
+					undefined,
+					2
+				),
+				dir: DIRECTORIES.TEST_CHECKS_INTEGRATION
+			}
+		];
+		filesToGenerate = filesToGenerate.concat(files);
+	}
+
+	if (GET_IS_RULE_MATCHES) {
+		const ruleMatchesJs = {
+			name: `${ruleName}-matches.js`,
+			content: `
+			// TODO: Filter node(s)
+			return node;
+			`,
+			dir: DIRECTORIES.RULES
+		};
+		filesToGenerate = filesToGenerate.concat(ruleMatchesJs);
+		const ruleMatchesTestJs = {
+			name: `${ruleName}-matches.js`,
+			content: `
+			describe('${ruleName}-matches', function() {
+				'use strict';
+				// TODO: Write tests
+			})
+			`,
+			dir: DIRECTORIES.TEST_RULES_MATCHES
+		};
+		filesToGenerate = filesToGenerate.concat(ruleMatchesTestJs);
+	}
+
+	const checkFiles = ruleChecks
+		.map(c => {
+			const cName = c.GET_CHECK_NAME.toLowerCase();
+			const outDir = `${DIRECTORIES.CHECKS}/${c.GET_CHECK_CATEGORY}/`;
+			const outTestDir = `${DIRECTORIES.CHECKS}/${c.GET_CHECK_CATEGORY}/`;
+			const checkJson = {
+				name: `${cName}.json`,
+				content: JSON.stringify(
+					{
+						id: `${cName}`,
+						evaluate: `${cName}.js`,
+						metadata: {
+							impact: '',
+							messages: {
+								pass: '',
+								fail: '',
+								incomplete: ''
+							}
+						}
+					},
+					undefined,
+					2
+				),
+				dir: outDir
+			};
+			const checkJs = {
+				name: `${cName}.js`,
+				content: `
+				// TODO: Logic for check
+				return true;
+				`,
+				dir: outDir
+			};
+			const checkTestJs = {
+				name: `${cName}.js`,
+				content: `
+				describe('${cName} tests', function() {
+					'use strict';
+					// TODO: Write tests
+				})
+				`,
+				dir: outTestDir
+			};
+			const files = [checkJson, checkJs];
+			if (GET_IS_UNIT_TEST_ASSETS) {
+				files.concat(checkTestJs);
+			}
+			return files;
+		})
+		.reduce((out, item) => {
+			out = out.concat(item);
+			return out;
+		}, []);
+	filesToGenerate = filesToGenerate.concat(checkFiles);
+	return filesToGenerate;
+};
+
+const createFile = fileMeta => {
+	const filePath = `${fileMeta.dir}/${fileMeta.name}`;
+	return new Promise((resolve, reject) => {
+		fs.writeFile(filePath, fileMeta.content, err => {
+			if (err) {
+				reject(err);
+			}
+			resolve(filePath);
+		});
+	});
+};
+
+let axeRulesFileNames;
+let axeRulesIds;
+let axeChecksFileNames;
+let axeChecksIds;
+
+const run = async () => {
+	// ensure axe exists, if not build axe, then start the generator
+	const axeExists = fs.existsSync(axeFile);
+	if (!axeExists) {
+		console.log(
+			chalk.black.bgRed.bold(
+				`Axe does not exist.Triggering build using - 'npm run build'.Rule Generator will continue after build.`
+			)
+		);
+		await cmd.run('npm run build');
+	}
+
+	// compute essentials
+	// 1. Get rule SPEC json file names
+	const ruleJsonSpecFiles = await filehound
```

Comment:
Consider replacing `filehound` with [`globby`](https://www.npmjs.com/package/globby). Globby has a much more familiar API and doesn't require you to read its documentation to figure out how to use it; just give it a glob pattern and you're good to go.